TrackingTools/TrackAssociator/src/TrackDetectorAssociator.cc

```diff
@@ -666,7 +666,6 @@ void TrackDetectorAssociator::getTAMuonChamberMatches(std::vector<TAMuonChamberM
       float halfWidthAtYPrime = 0.5f * narrowWidth + yPrime * tangent;
       distanceX = std::abs(localPoint.x()) - halfWidthAtYPrime;
       distanceY = std::abs(localPoint.y() - yCOWPOffset) - 0.5f * length;
-    } else if (const GEMChamber* gemchamber = dynamic_cast<const GEMChamber*>(geomDet)) {
```

Comment:
Since I am noticing, let me point out that this `else if` was not useless, instead.

With it, if `(const GEMChamber* gemchamber = dynamic_cast<const GEMChamber*>(geomDet))` the flux can avoid [L681](https://github.com/cms-sw/cmssw/pull/40554/files#diff-37c203e4484cc18140f603ae65ca3e8721e9a246df9244cc3f9de3990bee737aR681)

Now that you removed id, even if `(const GEMChamber* gemchamber = dynamic_cast<const GEMChamber*>(geomDet))` what is coded after `else` is executed

Reply:
Hi @perrotta, previously @Jo100sun had the GEMChamber and GEMSuperChamber cases separately, but I asked him to merge since the two branches were doing the same thing. Now, the new `else if` at 669 should take care of both cases. I think he just missed removing the old else if for GEMChamber  in the commit after merging the two cases.